instaloader/structures.py

```diff
@@ -132,7 +132,10 @@ def __hash__(self) -> int:
     def _obtain_metadata(self):
         if not self._full_metadata_dict:
             pic_json = self._context.get_json("p/{0}/".format(self.shortcode), params={})
-            self._full_metadata_dict = pic_json['entry_data']['PostPage'][0]['graphql']['shortcode_media']
+            if 'graphql' in pic_json:
+                self._full_metadata_dict = pic_json['graphql']['shortcode_media']
+            else:
+                self._full_metadata_dict = pic_json['entry_data']['PostPage'][0]['graphql']['shortcode_media']
             self._rhx_gis_str = pic_json.get('rhx_gis')
```

Comment:
This would evaluate to `None` with your proposed change.

Reply:
Well, I will test it Tomorrow evening. Here in Germany it's to late now, I must work tomorrow, so... but yeah I will look into it!